File: 00_bigO/nemo.js
let fish = ["nemo", "shark", "clown fish"];

fish = new Array(100000).fill("nemo");

const start = performance.now();

function findNemo(arr) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] == "nemo") {
      console.log("We found nemo");
    } else {
      console.log("Oops not nemo");
    }
  }
}

findNemo(fish);

const end = performance.now();

console.log(`Time taken: ${(end - start).toFixed(12)} ms`);

File: 00_bigO/nemo.py
import time
fish = ['nemo', 'shart', 'clown fish']

fish = ['nemo']*100000

start = time.time()


def find_nemo(arr):
    for i in arr:
        if i == 'nemo':
            print('We found nemo')
        else:
            print('Oops not nemo')


find_nemo(fish)

end = time.time()

print(f'Time taken: {(end - start) * 1000:.10f} ms')
